src/mavedb/routers/experiment_sets.py

```diff
@@ -23,6 +25,40 @@
 logger = logging.getLogger(__name__)
 
 
+@router.get(
+    "/check-authorizations/{urn}",
+    status_code=200,
+    response_model=bool
+)
+async def check_experiment_set_authorization(
+    *,
+    urn: str,
+    db: Session = Depends(deps.get_db),
+    user_data: UserData = Depends(get_current_user),
+) -> bool:
```

Comment:
Something I think would work really well here is leveraging the ability of Pydantic to parse enum values from a request (see: https://fastapi.tiangolo.com/tutorial/path-params/#predefined-values) and our existing `Action` enum from the permissions module. That will allow us to check the authorizations on a given resource at the given action level and make this request a little more generic without adding multiple functions.

The only thing is we need to change the way the Action enum is defined:
```
class Action(Enum):
    READ = "read"
    UPDATE = "update"
    DELETE = "delete"
    ADD_EXPERIMENT = "add_experiment"
    ADD_SCORE_SET = "add_score_set"
    SET_SCORES = "set_scores"
    ADD_ROLE = "add_role"
    PUBLISH = "publish"
```

```suggestion
@router.get(
    "/user-is-authorized/{urn}/{action}",
    status_code=200,
    response_model=bool
)
async def check_experiment_set_authorization(
    *,
    urn: str,
    action: Action,
    db: Session = Depends(deps.get_db),
    user_data: UserData = Depends(get_current_user),
) -> bool:
```

With this definition, we'll be able to make distinct requests like `/user-is-authorized/urn:00000-1/add_experiment` to check if the user can add an experiment and `/user-is-authorized/urn:00000-1/publish` to check if the user can publish. This might not be explicitly needed for this feature, but in my mind it's a nice generic way to write this endpoint that would probably end up being useful in the future.

Reply:
Thanks Ben! You give me wonderful suggestions. I considered writing a generic function before but it seemed too complex cause I didn't know we have `has_permission` already, so I gave up. I'll write a new one based on your above suggestions.